### PR TITLE
fix a TODO in ValidatingAdmissionWebhook

### PR DESCRIPTION
**Pull request by CaoShuFeng**
Tests still passes without this filter.

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
/assign @sttts 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/validating/plugin_test.go [hunk 1]
```diff
@@ -47,11 +47,6 @@ func TestValidate(t *testing.T) {
 	defer close(stopCh)
 
 	for _, tt := range webhooktesting.NewTestCases(serverURL) {
-		// TODO: re-enable all tests
```
Comment (sttts) [inline staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/validating/plugin_test.go]: I remember that they didn't when I did the refactoring. Anyway, good to know that they do. I might have fixed the issue and forgot to remove this todo.

Comment (CaoShuFeng) [reply staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/validating/plugin_test.go]: Hi, @sttts , I rebased this.
All test cases that return a [Patch](https://github.com/kubernetes/kubernetes/blob/b005f2fba340e2a9a3111b499f4f0eaf1ceed3eb/pkg/apis/admission/types.go#L91) is marked as MutationSpecific and they will not be test in validating webhook here.

This is a normal use case, I think.
Please take a look, thanks.

Comment (sttts) [reply staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/validating/plugin_test.go]: Wouldn't it be cleaner to split up `NewTestCase` into mutating and non-mutating tests?

Comment (CaoShuFeng) [reply staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/validating/plugin_test.go]: Done.

**Comment by CaoShuFeng** [comment]
Hi, @sttts
Can you label this again?
Thanks

**Pull request merged**